machine_test.go

```diff
@@ -40,12 +41,24 @@ func (s *fakeSupervisor) Setargs(ctx context.Context, args []string, _ *struct{}
 	return nil
 }
 
-func (s *fakeSupervisor) Exec(ctx context.Context, exec io.Reader, _ *struct{}) error {
-	var err error
-	s.Image, err = ioutil.ReadAll(exec)
+func (s *fakeSupervisor) Setbinary(ctx context.Context, binary io.Reader, _ *struct{}) (err error) {
```

Comment:
```suggestion
func (s *fakeSupervisor) SetBinary(ctx context.Context, binary io.Reader, _ *struct{}) (err error) {
```

I guess `Setbinary` was meant to mimic `Setenv`, but I think `Setenv` is just an homage to the C standard library 😄 .